doc/cprover-manual/contracts-assigns.md

```diff
@@ -17,28 +17,46 @@ value(s) therein are not modified.
 
 ### Object slice expressions
 
-The following functions can be used in assigns clause to specify ranges of 
-assignable addresses.
+The following functions can be used in assigns clauses to specify ranges of assignable bytes.
 
-Given a pointer `ptr` pointing into some object `o`, `__CPROVER_object_from(ptr)` 
-specifies that all bytes starting from the given pointer and until the end of 
-the object are assignable:
+Given an lvalue expression `expr` with a complete type `expr_t`,
+ `__CPROVER_typed_target(expr)` specifies that the range
+ of `sizeof(expr_t)` bytes starting at `&expr` is assignable:
 ```c
-__CPROVER_size_t __CPROVER_object_from(void *ptr); 
+__CPROVER_assignable_t __CPROVER_typed_target(expr_t expr);
```

Comment:
They are not equivalent:

`__CPROVER_assigns(__CPROVER_typed_target(b))` says the range of bytes starting at `&b` up to `sizeof(b)` is assignable, which is basically the pointer `b` itself.

`__CPROVER_assigns(*b)` says the range of bytes starting at `&(*b)` up to `sizeof(*b)` is assignable, so that is the struct.

Reply:
The documentation gives the start address and the size of the range in bytes for each builtin operator, what additional clarification do you think we could add ?